### DISCUSSION TITLE
Intune Deployment 2.0

### DISCUSSION THREAD

**Discussion by jNizM** [Romanitho/Winget-AutoUpdate#735] [Q&A]
Hey guys.

Can we get a new guide to deploy WAU 2.0 with Intune?

**Comment by Romanitho** [comment]
you don't have that much to do. Package your MSI as usual, and manage it with the policies from ADMX.
Example:
`msiexec /i WAU.msi /qn USEWHITELIST=1 UPDATESATLOGON=0 UPDATESINTERVAL=Weekly `

**Reply by jNizM** [reply]
WAU what was previously installed with `.intunewin`.

After the WAU autoupdate to version 2.0 I had two installation entries in `appwiz.cpl` with the old and new icon of WAU.

In your release notes you write the following:
`WAU.msi will update in the current install folder. It is recommanded to uninstall WAU v1 and reinstall WAU v2 in default install folder (ProgramFiles).`

After I had two versions, I switched the previously rolled out package to uninstall in Intune.

After uninstalling, the WAU folder under ProgramData is empty except for the log folder, but I still have an entry for WAU 2.0 (with a new icon) in appwiz.

According to the registry, `HKLM:SOFTWARE\Romanitho\Winget-AutoUpdate` is still set and therefore the program is displayed as installed in appwiz, although it is no longer present.

However, the old registry key `HKLM:SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Winget-AutoUpdate` has been successfully deleted.

**Reply by Romanitho** [reply]
Just a question: Are you still deploying the old version with Intune? It can happen that the auto-update installs the MSI version, and Intune detects that WAU v1 is missing (if your detection method is based on the registry). Then it tries to reinstall v1, which could lead to double entries in the appwiz.cpl.

**Reply by jNizM** [reply]
As soon as I saw that v2 was installed, I went to the Github page and saw the entry in the release notes.
I then set the deployment to uninstall to re-roll out v2 later.

**Reply by jNizM** [reply]
It looks like before I re-roll out v2, I need to clean up the registry first so that the deployment of v2 works smoothly, as v2 is shown as installed even when it is not.

**Reply by Romanitho** [reply]
It should not change anymore. We are now adhering to classic MSI standards, so this kind of issue shouldn't occur again in the future. Apologies for the inconvenience.
You can also uninstall V2, reinstall the latest V1, uninstall it properly, and then install V2 again.

**Reply by jNizM** [reply]
All good. Nothing you can't do with a remidations script.

Btw. Should we pack the msi as intune app or can we "Line-of-business app" deployment for the msi.
![image](https://github.com/user-attachments/assets/e79ab5be-f910-4c21-835b-2f36cd8366ee)


I am not sure about the best practice

edit:
looks like I do it again with intunewin. do you provide an msiexec uninstall command (guid / uuid)?
(https://patchmypc.com/intune-line-of-business-apps-vs-win32-apps)

**Reply by Romanitho** [reply]
I always use Win32 even for MSI.
regarding uninstall, I just configure the MSI /x with the MSI guid.

**Reply by jNizM** [reply]
So is this the correct way to install this with GPO + just updates on logon?

`msiexec /i "WAU.msi" /qn USERCONTEXT=1 LISTPATH=GPO DESKTOPSHORTCUT=0 STARTMENUSHORTCUT=0 NOTIFICATIONLEVEL=SuccessOnly UPDATESATLOGON=1`

**Reply by jNizM** [reply]
Looks like this is working so far

![image](https://github.com/user-attachments/assets/e3a725be-377d-4715-a079-50cc6794d11c)

**Reply by Romanitho** [reply]
Seems good.

**Reply by NRL-LEastham** [reply]
> I always use Win32 even for MSI. regarding uninstall, I just configure the MSI /x with the MSI guid.

This only works well if the MSI GUID remains the same each version. Unfortuantely I've discovered that between 2.0.0 and 2.0.1 the GUID changed so I guess this will keep changing with each new release. This really messes with things like Intune where you need the GUID to be static an then compare the version number to know when to update, if you're doing Intune properly and adding new packages to superceded the old ones. Getting the GUID prior to the next version install is bit of a pain but thankfully i found some PS code to do that.

**Comment by UnhappyPoetry** [comment]
I think I have the same situation. I have the old WAU entries and the new one. 
How did you uninstall V1 on your clients?

My Uninstall command is:
"%systemroot%\sysnative\WindowsPowerShell\v1.0\powershell.exe" -noprofile -executionpolicy bypass -file "Winget-AutoUpdate-Install.ps1" -Silent -Uninstall"

And I'm using the Registry detection method "HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Windows\CurrentVersion\Uninstall\Winget-AutoUpdate"

It seems like he doesn' even try to uninstall V1.

**Reply by jNizM** [reply]
This is my v1 App and it works for me

![image](https://github.com/user-attachments/assets/38500713-e0aa-4b7b-a8d5-c6ebbd9ad8a9)

Assignments
-> Uninstall All Devices

**Reply by jNizM** [reply]
And this is my short cleanup script after uninstallation of v1 and v2 installation.
It deletes the old v1 folder (contains logs)

```powershell
$OldPath = "C:\ProgramData\Winget-AutoUpdate"
$NewPath = "C:\Program Files\Winget-AutoUpdate"
$RegPath = "HKLM:\SOFTWARE\Romanitho"

if (Test-Path -Path $NewPath)
{
    if (Test-Path -Path $RegPath)
    {
        if (Test-Path -Path $OldPath)
        {
            Remove-Item -Path $OldPath -Recurse
        }
        else
        {
            exit 0
        }
    }
    else
    {
        exit 0
    }
}
else
{
	exit 0
}
```

**Reply by UnhappyPoetry** [reply]
Thank you!! I'll compare it and give it a try.
